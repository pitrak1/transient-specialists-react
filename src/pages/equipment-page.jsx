import React from 'react'
import {
  Button,
  CircularProgress,
  TextField,
  Toolbar,
  Typography,
} from '@material-ui/core'
import FullTable from '../components/full-table.jsx'
import { withRouter } from 'react-router'
import api from '../api.js'

export class EquipmentPage extends React.Component {
  state = {
    alert: null,
    ascending: true,
    data: {},
    error: null,
    loading: true,
    page: 0,
    perPage: 10,
    searchValue: this.props.match.params.search || '',
    sortBy: 'serialNumber',
  }

  componentDidMount() {
    this.getData()
  }

  getData = () => {
    this.setState({ loading: true })
    const { ascending, page, perPage, searchValue, sortBy } = this.state
    api.getIndex(
      'equipment',
      {
        ascending,
        page,
        perPage,
        searchValue,
        sortBy,
      },
      result => {
        this.setState({ loading: false, data: result })
      },
      error => {
        this.setState({ loading: false, error })
      },
    )
  }

  handleDetailsClick = id => {
    this.props.history.push(`/equipment/${id}`)
  }

  handleAddClick = _event => {
    this.props.history.push(`/equipment/create`)
  }

  handleSearchChange = event => {
    this.setState({ searchValue: event.target.value })
  }

  handleSearchClick = () => {
    this.getData()
  }

  handlePageChange = (_event, newPage) => {
    this.setState({ page: newPage }, this.getData)
  }

  handlePerPageChange = event => {
    this.setState({ perPage: event.target.value }, this.getData)
  }

  handleSort = (sortBy, ascending) => {
    this.setState({ sortBy, ascending }, this.getData)
  }

  render() {
    if (this.state.loading) {
      return <CircularProgress size={120} />
    }

    if (this.state.error) {
      return <div>{this.state.error}</div>
    }

    const headers = [
      { type: 'value', id: 'serialNumber', label: 'Serial Number' },
      { type: 'value', id: 'oemName', label: 'OEM' },
      { type: 'value', id: 'modelName', label: 'Model' },
      { type: 'value', id: 'typeName', label: 'Type' },
      { type: 'value', id: 'notes', label: 'Notes' },
      { type: 'value', id: 'eventStatus', label: 'Status' },
      { type: 'value', id: 'eventCompanyNotes', label: 'Company/Notes' },
      { type: 'value', id: 'eventStartDate', label: 'Start Date' },
      { type: 'value', id: 'eventEndDate', label: 'End Date' },
      { type: 'button', id: 'details' },
    ]

    const data = this.state.data.equipment.map(equipment => ({
      id: equipment.id,
      cells: [
        { id: 'serialNumber', type: 'value', value: equipment.serialNumber },
        { id: 'oemName', type: 'value', value: equipment.oemName },
        { id: 'modelName', type: 'value', value: equipment.modelName },
        { id: 'typeName', type: 'value', value: equipment.typeName },
        { id: 'notes', type: 'value', value: equipment.notes },
        { id: 'eventStatus', type: 'value', value: equipment.eventStatus },
        {
          id: 'eventCompanyNotes',
          type: 'value',
          value: equipment.eventCompanyNotes,
        },
        {
          id: 'eventStartDate',
          type: 'value',
          value: equipment.eventStartDate,
        },
        { id: 'eventEndDate', type: 'value', value: equipment.eventEndDate },
        {
          id: 'details',
          type: 'button',
          value: 'Details',
          callback: this.handleDetailsClick,
        },
      ],
    }))

    return (
      <div>
        {this.state.alert && <div>{this.state.alert}</div>}
        <Toolbar>
          <Typography variant='h5'>Equipment</Typography>
          <Button onClick={this.handleAddClick}>Add</Button>
          <div style={{ flexGrow: 1 }}></div>
          <TextField
            id='search'
            label='Search'
            value={this.state.searchValue}
            onChange={this.handleSearchChange}
          />
          <Button onClick={this.handleSearchClick}>Search</Button>
        </Toolbar>
        <FullTable
          ascending={this.state.ascending}
          count={this.state.data.count}
          data={data}
          headers={headers}
          onPageChange={this.handlePageChange}
          onPerPageChange={this.handlePerPageChange}
          onSort={this.handleSort}
          page={this.state.page}
          perPage={this.state.perPage}
          perPageOptions={[5, 10, 25]}
          sortBy={this.state.sortBy}
        />
      </div>
    )
  }
}

export default withRouter(EquipmentPage)
