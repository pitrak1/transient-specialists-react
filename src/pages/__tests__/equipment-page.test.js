import React from 'react'
import { EquipmentPage } from '../equipment-page'
import { shallow } from 'enzyme'
import sinon from 'sinon'
import api from '../../api.js'

describe('EquipmentPage', () => {
  const data = [
    {
      id: 1,
      serialNumber: 'Equipment 1',
      oemName: 'OEM 3',
      modelName: 'Model 1',
      typeName: 'Type 2',
    },
    {
      id: 2,
      serialNumber: 'Equipment 2',
      oemName: 'OEM 1',
      modelName: 'Model 3',
      typeName: 'Type 3',
    },
    {
      id: 3,
      serialNumber: 'Equipment 3',
      oemName: 'OEM 2',
      modelName: 'Model 2',
      typeName: 'Type 1',
    },
  ]

  it('renders spinner if not yet loaded', () => {
    const node = shallow(<EquipmentPage match={{ params: {} }} />)
    expect(node).toMatchSnapshot()
  })

  it('renders an error if load fails', () => {
    api.getIndex = sinon.stub().callsArgWith(2, 'some error')
    const node = shallow(<EquipmentPage match={{ params: {} }} />)
    expect(node).toMatchSnapshot()
  })

  it('renders data if load succeeds', () => {
    api.getIndex = sinon.stub().callsArgWith(1, data)
    const node = shallow(<EquipmentPage match={{ params: {} }} />)
    expect(node).toMatchSnapshot()
  })

  it('sorts data ascending when column header is clicked', () => {
    api.getIndex = sinon.stub().callsArgWith(1, data)
    const node = shallow(<EquipmentPage match={{ params: {} }} />)
    node.instance().handleSort(null, { id: 'oemName' })
    expect(node).toMatchSnapshot()
  })

  it('sorts data descending when sorted column header is clicked', () => {
    api.getIndex = sinon.stub().callsArgWith(1, data)
    const node = shallow(<EquipmentPage match={{ params: {} }} />)
    node.instance().handleSort(null, { id: 'oemName' })
    node.instance().handleSort(null, { id: 'oemName' })
    expect(node).toMatchSnapshot()
  })

  it('filters data based on serial number', () => {
    api.getIndex = sinon.stub().callsArgWith(1, data)
    const node = shallow(<EquipmentPage match={{ params: {} }} />)
    node.instance().handleSearchChange(null, 'Equipment 1')
    expect(node).toMatchSnapshot()
  })

  it('filters data based on OEM name', () => {
    api.getIndex = sinon.stub().callsArgWith(1, data)
    const node = shallow(<EquipmentPage match={{ params: {} }} />)
    node.instance().handleSearchChange(null, 'OEM 1')
    expect(node).toMatchSnapshot()
  })

  it('filters data based on model name', () => {
    api.getIndex = sinon.stub().callsArgWith(1, data)
    const node = shallow(<EquipmentPage match={{ params: {} }} />)
    node.instance().handleSearchChange(null, 'Model 1')
    expect(node).toMatchSnapshot()
  })

  it('filters data based on type name', () => {
    api.getIndex = sinon.stub().callsArgWith(1, data)
    const node = shallow(<EquipmentPage match={{ params: {} }} />)
    node.instance().handleSearchChange(null, 'Type 1')
    expect(node).toMatchSnapshot()
  })
})
