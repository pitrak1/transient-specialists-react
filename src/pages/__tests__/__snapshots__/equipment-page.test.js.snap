// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EquipmentPage filters data based on OEM name 1`] = `
<div>
  <Flex
    as="span"
    direction="row"
    elementRef={[Function]}
    inline={false}
    justifyItems="start"
    visualDebug={false}
    wrapItems={false}
  >
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={true}
      shrink={true}
    >
      <Heading
        border="none"
        color="inherit"
        ellipsis={false}
        level="h1"
        margin="medium"
      >
        Equipment
      </Heading>
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      margin="small"
      shrink={false}
    >
      <TextInput
        display="block"
        inputContainerRef={[Function]}
        inputRef={[Function]}
        interaction="enabled"
        isRequired={false}
        messages={Array []}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        renderLabel="Search"
        size="medium"
        textAlign="start"
        type="text"
        value="OEM 1"
      />
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      shrink={false}
    >
      <Button
        as="button"
        buttonRef={[Function]}
        cursor="pointer"
        fluidWidth={false}
        margin="0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="default"
      >
        Add Equipment
      </Button>
    </FlexItem>
  </Flex>
  <Table
    caption="Equipment"
    hover={true}
    layout="auto"
  >
    <Head>
      <Row>
        <ColHeader
          id="serialNumber"
          onRequestSort={[Function]}
          sortDirection="ascending"
          textAlign="start"
        >
          Serial Number
        </ColHeader>
        <ColHeader
          id="oemName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          OEM Name
        </ColHeader>
        <ColHeader
          id="modelName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Model Name
        </ColHeader>
        <ColHeader
          id="typeName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Type Name
        </ColHeader>
        <ColHeader
          id="Details"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
      </Row>
    </Head>
    <Body>
      <Row
        key="2"
      >
        <Cell
          textAlign="start"
        >
          Equipment 2
        </Cell>
        <Cell
          textAlign="start"
        >
          OEM 1
        </Cell>
        <Cell
          textAlign="start"
        >
          Model 3
        </Cell>
        <Cell
          textAlign="start"
        >
          Type 3
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={2}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Details
          </Button>
        </Cell>
      </Row>
    </Body>
  </Table>
</div>
`;

exports[`EquipmentPage filters data based on model name 1`] = `
<div>
  <Flex
    as="span"
    direction="row"
    elementRef={[Function]}
    inline={false}
    justifyItems="start"
    visualDebug={false}
    wrapItems={false}
  >
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={true}
      shrink={true}
    >
      <Heading
        border="none"
        color="inherit"
        ellipsis={false}
        level="h1"
        margin="medium"
      >
        Equipment
      </Heading>
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      margin="small"
      shrink={false}
    >
      <TextInput
        display="block"
        inputContainerRef={[Function]}
        inputRef={[Function]}
        interaction="enabled"
        isRequired={false}
        messages={Array []}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        renderLabel="Search"
        size="medium"
        textAlign="start"
        type="text"
        value="Model 1"
      />
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      shrink={false}
    >
      <Button
        as="button"
        buttonRef={[Function]}
        cursor="pointer"
        fluidWidth={false}
        margin="0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="default"
      >
        Add Equipment
      </Button>
    </FlexItem>
  </Flex>
  <Table
    caption="Equipment"
    hover={true}
    layout="auto"
  >
    <Head>
      <Row>
        <ColHeader
          id="serialNumber"
          onRequestSort={[Function]}
          sortDirection="ascending"
          textAlign="start"
        >
          Serial Number
        </ColHeader>
        <ColHeader
          id="oemName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          OEM Name
        </ColHeader>
        <ColHeader
          id="modelName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Model Name
        </ColHeader>
        <ColHeader
          id="typeName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Type Name
        </ColHeader>
        <ColHeader
          id="Details"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
      </Row>
    </Head>
    <Body>
      <Row
        key="1"
      >
        <Cell
          textAlign="start"
        >
          Equipment 1
        </Cell>
        <Cell
          textAlign="start"
        >
          OEM 3
        </Cell>
        <Cell
          textAlign="start"
        >
          Model 1
        </Cell>
        <Cell
          textAlign="start"
        >
          Type 2
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={1}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Details
          </Button>
        </Cell>
      </Row>
    </Body>
  </Table>
</div>
`;

exports[`EquipmentPage filters data based on serial number 1`] = `
<div>
  <Flex
    as="span"
    direction="row"
    elementRef={[Function]}
    inline={false}
    justifyItems="start"
    visualDebug={false}
    wrapItems={false}
  >
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={true}
      shrink={true}
    >
      <Heading
        border="none"
        color="inherit"
        ellipsis={false}
        level="h1"
        margin="medium"
      >
        Equipment
      </Heading>
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      margin="small"
      shrink={false}
    >
      <TextInput
        display="block"
        inputContainerRef={[Function]}
        inputRef={[Function]}
        interaction="enabled"
        isRequired={false}
        messages={Array []}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        renderLabel="Search"
        size="medium"
        textAlign="start"
        type="text"
        value="Equipment 1"
      />
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      shrink={false}
    >
      <Button
        as="button"
        buttonRef={[Function]}
        cursor="pointer"
        fluidWidth={false}
        margin="0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="default"
      >
        Add Equipment
      </Button>
    </FlexItem>
  </Flex>
  <Table
    caption="Equipment"
    hover={true}
    layout="auto"
  >
    <Head>
      <Row>
        <ColHeader
          id="serialNumber"
          onRequestSort={[Function]}
          sortDirection="ascending"
          textAlign="start"
        >
          Serial Number
        </ColHeader>
        <ColHeader
          id="oemName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          OEM Name
        </ColHeader>
        <ColHeader
          id="modelName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Model Name
        </ColHeader>
        <ColHeader
          id="typeName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Type Name
        </ColHeader>
        <ColHeader
          id="Details"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
      </Row>
    </Head>
    <Body>
      <Row
        key="1"
      >
        <Cell
          textAlign="start"
        >
          Equipment 1
        </Cell>
        <Cell
          textAlign="start"
        >
          OEM 3
        </Cell>
        <Cell
          textAlign="start"
        >
          Model 1
        </Cell>
        <Cell
          textAlign="start"
        >
          Type 2
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={1}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Details
          </Button>
        </Cell>
      </Row>
    </Body>
  </Table>
</div>
`;

exports[`EquipmentPage filters data based on type name 1`] = `
<div>
  <Flex
    as="span"
    direction="row"
    elementRef={[Function]}
    inline={false}
    justifyItems="start"
    visualDebug={false}
    wrapItems={false}
  >
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={true}
      shrink={true}
    >
      <Heading
        border="none"
        color="inherit"
        ellipsis={false}
        level="h1"
        margin="medium"
      >
        Equipment
      </Heading>
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      margin="small"
      shrink={false}
    >
      <TextInput
        display="block"
        inputContainerRef={[Function]}
        inputRef={[Function]}
        interaction="enabled"
        isRequired={false}
        messages={Array []}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        renderLabel="Search"
        size="medium"
        textAlign="start"
        type="text"
        value="Type 1"
      />
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      shrink={false}
    >
      <Button
        as="button"
        buttonRef={[Function]}
        cursor="pointer"
        fluidWidth={false}
        margin="0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="default"
      >
        Add Equipment
      </Button>
    </FlexItem>
  </Flex>
  <Table
    caption="Equipment"
    hover={true}
    layout="auto"
  >
    <Head>
      <Row>
        <ColHeader
          id="serialNumber"
          onRequestSort={[Function]}
          sortDirection="ascending"
          textAlign="start"
        >
          Serial Number
        </ColHeader>
        <ColHeader
          id="oemName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          OEM Name
        </ColHeader>
        <ColHeader
          id="modelName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Model Name
        </ColHeader>
        <ColHeader
          id="typeName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Type Name
        </ColHeader>
        <ColHeader
          id="Details"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
      </Row>
    </Head>
    <Body>
      <Row
        key="3"
      >
        <Cell
          textAlign="start"
        >
          Equipment 3
        </Cell>
        <Cell
          textAlign="start"
        >
          OEM 2
        </Cell>
        <Cell
          textAlign="start"
        >
          Model 2
        </Cell>
        <Cell
          textAlign="start"
        >
          Type 1
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={3}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Details
          </Button>
        </Cell>
      </Row>
    </Body>
  </Table>
</div>
`;

exports[`EquipmentPage renders an error if load fails 1`] = `
<Alert
  liveRegionPoliteness="assertive"
  margin="x-small 0"
  open={true}
  screenReaderOnly={false}
  timeout={0}
  transition="fade"
  variant="error"
>
  some error
</Alert>
`;

exports[`EquipmentPage renders data if load succeeds 1`] = `
<div>
  <Flex
    as="span"
    direction="row"
    elementRef={[Function]}
    inline={false}
    justifyItems="start"
    visualDebug={false}
    wrapItems={false}
  >
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={true}
      shrink={true}
    >
      <Heading
        border="none"
        color="inherit"
        ellipsis={false}
        level="h1"
        margin="medium"
      >
        Equipment
      </Heading>
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      margin="small"
      shrink={false}
    >
      <TextInput
        display="block"
        inputContainerRef={[Function]}
        inputRef={[Function]}
        interaction="enabled"
        isRequired={false}
        messages={Array []}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        renderLabel="Search"
        size="medium"
        textAlign="start"
        type="text"
        value=""
      />
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      shrink={false}
    >
      <Button
        as="button"
        buttonRef={[Function]}
        cursor="pointer"
        fluidWidth={false}
        margin="0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="default"
      >
        Add Equipment
      </Button>
    </FlexItem>
  </Flex>
  <Table
    caption="Equipment"
    hover={true}
    layout="auto"
  >
    <Head>
      <Row>
        <ColHeader
          id="serialNumber"
          onRequestSort={[Function]}
          sortDirection="ascending"
          textAlign="start"
        >
          Serial Number
        </ColHeader>
        <ColHeader
          id="oemName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          OEM Name
        </ColHeader>
        <ColHeader
          id="modelName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Model Name
        </ColHeader>
        <ColHeader
          id="typeName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Type Name
        </ColHeader>
        <ColHeader
          id="Details"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
      </Row>
    </Head>
    <Body>
      <Row
        key="1"
      >
        <Cell
          textAlign="start"
        >
          Equipment 1
        </Cell>
        <Cell
          textAlign="start"
        >
          OEM 3
        </Cell>
        <Cell
          textAlign="start"
        >
          Model 1
        </Cell>
        <Cell
          textAlign="start"
        >
          Type 2
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={1}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Details
          </Button>
        </Cell>
      </Row>
      <Row
        key="2"
      >
        <Cell
          textAlign="start"
        >
          Equipment 2
        </Cell>
        <Cell
          textAlign="start"
        >
          OEM 1
        </Cell>
        <Cell
          textAlign="start"
        >
          Model 3
        </Cell>
        <Cell
          textAlign="start"
        >
          Type 3
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={2}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Details
          </Button>
        </Cell>
      </Row>
      <Row
        key="3"
      >
        <Cell
          textAlign="start"
        >
          Equipment 3
        </Cell>
        <Cell
          textAlign="start"
        >
          OEM 2
        </Cell>
        <Cell
          textAlign="start"
        >
          Model 2
        </Cell>
        <Cell
          textAlign="start"
        >
          Type 1
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={3}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Details
          </Button>
        </Cell>
      </Row>
    </Body>
  </Table>
</div>
`;

exports[`EquipmentPage renders spinner if not yet loaded 1`] = `
<Spinner
  as="div"
  renderTitle="Loading"
  size="large"
  variant="default"
/>
`;

exports[`EquipmentPage sorts data ascending when column header is clicked 1`] = `
<div>
  <Flex
    as="span"
    direction="row"
    elementRef={[Function]}
    inline={false}
    justifyItems="start"
    visualDebug={false}
    wrapItems={false}
  >
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={true}
      shrink={true}
    >
      <Heading
        border="none"
        color="inherit"
        ellipsis={false}
        level="h1"
        margin="medium"
      >
        Equipment
      </Heading>
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      margin="small"
      shrink={false}
    >
      <TextInput
        display="block"
        inputContainerRef={[Function]}
        inputRef={[Function]}
        interaction="enabled"
        isRequired={false}
        messages={Array []}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        renderLabel="Search"
        size="medium"
        textAlign="start"
        type="text"
        value=""
      />
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      shrink={false}
    >
      <Button
        as="button"
        buttonRef={[Function]}
        cursor="pointer"
        fluidWidth={false}
        margin="0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="default"
      >
        Add Equipment
      </Button>
    </FlexItem>
  </Flex>
  <Table
    caption="Equipment"
    hover={true}
    layout="auto"
  >
    <Head>
      <Row>
        <ColHeader
          id="serialNumber"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Serial Number
        </ColHeader>
        <ColHeader
          id="oemName"
          onRequestSort={[Function]}
          sortDirection="ascending"
          textAlign="start"
        >
          OEM Name
        </ColHeader>
        <ColHeader
          id="modelName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Model Name
        </ColHeader>
        <ColHeader
          id="typeName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Type Name
        </ColHeader>
        <ColHeader
          id="Details"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
      </Row>
    </Head>
    <Body>
      <Row
        key="2"
      >
        <Cell
          textAlign="start"
        >
          Equipment 2
        </Cell>
        <Cell
          textAlign="start"
        >
          OEM 1
        </Cell>
        <Cell
          textAlign="start"
        >
          Model 3
        </Cell>
        <Cell
          textAlign="start"
        >
          Type 3
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={2}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Details
          </Button>
        </Cell>
      </Row>
      <Row
        key="3"
      >
        <Cell
          textAlign="start"
        >
          Equipment 3
        </Cell>
        <Cell
          textAlign="start"
        >
          OEM 2
        </Cell>
        <Cell
          textAlign="start"
        >
          Model 2
        </Cell>
        <Cell
          textAlign="start"
        >
          Type 1
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={3}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Details
          </Button>
        </Cell>
      </Row>
      <Row
        key="1"
      >
        <Cell
          textAlign="start"
        >
          Equipment 1
        </Cell>
        <Cell
          textAlign="start"
        >
          OEM 3
        </Cell>
        <Cell
          textAlign="start"
        >
          Model 1
        </Cell>
        <Cell
          textAlign="start"
        >
          Type 2
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={1}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Details
          </Button>
        </Cell>
      </Row>
    </Body>
  </Table>
</div>
`;

exports[`EquipmentPage sorts data descending when sorted column header is clicked 1`] = `
<div>
  <Flex
    as="span"
    direction="row"
    elementRef={[Function]}
    inline={false}
    justifyItems="start"
    visualDebug={false}
    wrapItems={false}
  >
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={true}
      shrink={true}
    >
      <Heading
        border="none"
        color="inherit"
        ellipsis={false}
        level="h1"
        margin="medium"
      >
        Equipment
      </Heading>
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      margin="small"
      shrink={false}
    >
      <TextInput
        display="block"
        inputContainerRef={[Function]}
        inputRef={[Function]}
        interaction="enabled"
        isRequired={false}
        messages={Array []}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        renderLabel="Search"
        size="medium"
        textAlign="start"
        type="text"
        value=""
      />
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      shrink={false}
    >
      <Button
        as="button"
        buttonRef={[Function]}
        cursor="pointer"
        fluidWidth={false}
        margin="0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="default"
      >
        Add Equipment
      </Button>
    </FlexItem>
  </Flex>
  <Table
    caption="Equipment"
    hover={true}
    layout="auto"
  >
    <Head>
      <Row>
        <ColHeader
          id="serialNumber"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Serial Number
        </ColHeader>
        <ColHeader
          id="oemName"
          onRequestSort={[Function]}
          sortDirection="descending"
          textAlign="start"
        >
          OEM Name
        </ColHeader>
        <ColHeader
          id="modelName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Model Name
        </ColHeader>
        <ColHeader
          id="typeName"
          onRequestSort={[Function]}
          sortDirection="none"
          textAlign="start"
        >
          Type Name
        </ColHeader>
        <ColHeader
          id="Details"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
      </Row>
    </Head>
    <Body>
      <Row
        key="1"
      >
        <Cell
          textAlign="start"
        >
          Equipment 1
        </Cell>
        <Cell
          textAlign="start"
        >
          OEM 3
        </Cell>
        <Cell
          textAlign="start"
        >
          Model 1
        </Cell>
        <Cell
          textAlign="start"
        >
          Type 2
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={1}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Details
          </Button>
        </Cell>
      </Row>
      <Row
        key="3"
      >
        <Cell
          textAlign="start"
        >
          Equipment 3
        </Cell>
        <Cell
          textAlign="start"
        >
          OEM 2
        </Cell>
        <Cell
          textAlign="start"
        >
          Model 2
        </Cell>
        <Cell
          textAlign="start"
        >
          Type 1
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={3}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Details
          </Button>
        </Cell>
      </Row>
      <Row
        key="2"
      >
        <Cell
          textAlign="start"
        >
          Equipment 2
        </Cell>
        <Cell
          textAlign="start"
        >
          OEM 1
        </Cell>
        <Cell
          textAlign="start"
        >
          Model 3
        </Cell>
        <Cell
          textAlign="start"
        >
          Type 3
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={2}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Details
          </Button>
        </Cell>
      </Row>
    </Body>
  </Table>
</div>
`;
